tests/Unit/Domain/Creators/Test_Sphere.cpp

```diff
@@ -784,42 +770,131 @@ void test_kerr_horizon_conforming(const bool use_time_dependence) {
+  domain::creators::Sphere::TimeDepOptionType time_dependent_options{};
+
+  // Set up theta phis
+  const size_t num_points = 10;
+  std::uniform_real_distribution<double> dist_phi{0., 2. * M_PI};
+  std::uniform_real_distribution<double> dist_theta{0., M_PI};
+  const std::array<DataVector, 2> theta_phi{
+      {make_with_random_values<DataVector>(gen, make_not_null(&dist_theta),
+                                           num_points),
+       make_with_random_values<DataVector>(gen, make_not_null(&dist_phi),
+                                           num_points)}};
+
+  const double time = 0.7;
+  const size_t l_max = 32;
+  const double mass = 0.8;
+  const std::array<double, 3> spin{{0.0, 0.0, 0.9}};
+  const double r_plus = mass * (1. + sqrt(1. - dot(spin, spin)));
+  const double inner_radius = r_plus;
+
+  const auto set_x = [](const DataVector& local_radius,
+                        const std::array<DataVector, 2>& local_theta_phi)
+      -> tnsr::I<DataVector, 3> {
+    tnsr::I<DataVector, 3> x{};
+    get<0>(x) = local_radius * sin(get<0>(local_theta_phi)) *
+                cos(get<1>(local_theta_phi));
+    get<1>(x) = local_radius * sin(get<0>(local_theta_phi)) *
+                sin(get<1>(local_theta_phi));
+    get<2>(x) = local_radius * cos(get<0>(local_theta_phi));
+    return x;
+  };
+
+  {
+    INFO(
+        "Check that inner radius is deformed to constant Boyer-Lindquist "
+        "radius");
+    const DataVector radius =
+        get(gr::Solutions::kerr_schild_radius_from_boyer_lindquist(
+            inner_radius, theta_phi, mass, spin));
+
+    // Set up coordinates on an ellipsoid of constant Boyer-Lindquist radius
+    const auto x = set_x(radius, theta_phi);
+
+    // Time dependence
+    time_dependent_options = std::make_unique<
+        domain::creators::time_dependence::Shape<domain::ObjectLabel::None>>(
+        time, l_max, mass, spin, std::array<double, 3>{{0., 0., 0.}},
+        inner_radius, 4.);
+
+    test_shape_distortion_general(time, std::move(time_dependent_options),
+                                  inner_radius, x);
+
+    // KerrSchild-BoyerLindquist. Use same x as above
+    time_dependent_options = domain::creators::sphere::TimeDependentMapOptions{
+        time,
+        {{l_max, domain::creators::time_dependent_options::
+                     KerrSchildFromBoyerLindquist{mass, spin}}},
+        std::nullopt,
+        std::nullopt,
+        std::nullopt};
+
+    test_shape_distortion_general(time, std::move(time_dependent_options),
+                                  inner_radius, x);
+  }
+
+  {
+    INFO("Check reading in Ylms from file");
+    const std::string h5_filename{"StrahlkorperCoefsFile.h5"};
+    const std::string subfile_name{"Ylm_coefs"};
+    if (file_system::check_if_file_exists(h5_filename)) {
+      file_system::rm(h5_filename, true);
+    }
+    ylm::Strahlkorper<Frame::Distorted> strahlkorper{l_max, inner_radius,
```

Comment:
This doesn't actually deform anything. Aren't all coefficients that get multiplied with -1 for the shape map zero? Just construct the strahlkorper from the Kerr-Schild radius you construct above, then all three tests here do the same deformation just in different ways.